plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueMetastoreModule.java

```diff
@@ -121,16 +132,12 @@ public static GlueCache createGlueCache(CachingHiveMetastoreConfig config, Catal
 
     @Provides
     @Singleton
-    public static GlueClient createGlueClient(GlueHiveMetastoreConfig config, OpenTelemetry openTelemetry)
+    public static GlueClient createGlueClient(GlueHiveMetastoreConfig config, @ForGlueHiveMetastore Set<ExecutionInterceptor> executionInterceptors)
     {
         GlueClientBuilder glue = GlueClient.builder();
 
         glue.overrideConfiguration(builder -> builder
-                .addExecutionInterceptor(AwsSdkTelemetry.builder(openTelemetry)
-                        .setCaptureExperimentalSpanAttributes(true)
-                        .setRecordIndividualHttpError(true)
-                        .build().newExecutionInterceptor())
-                .addExecutionInterceptor(new GlueHiveExecutionInterceptor(config.isSkipArchive()))
+                .executionInterceptors(ImmutableList.copyOf(executionInterceptors))
```

Comment:
You are loosing ordering here. Does it matter?

Reply:
> The set's iteration order is consistent with the binding order. This is convenient when multiple elements are contributed by the same module because that module can order its bindings appropriately. Avoid relying on the iteration order of elements contributed by different modules, since there is no equivalent mechanism to order modules.

So far order is maintained. If we keep adding new elements following the method above is safe. If we add an element in different module, then it could be added in random place (before or after this module). 

I have checked the existing interceptors and they are independent from each other, and I believe it is important thing for them. If order would matter then I guess it would have to work as a stack because you have methods there like `beforeX` and `afterX`.

Reply:
So I believe order does not matter here.